taichi/backends/dx/dx_device.cpp

```diff
@@ -4,6 +4,24 @@ namespace taichi {
 namespace lang {
 namespace directx11 {
 
+bool kD3d11DebugEnabled = false;  // D3D11 debugging is enabled. For testing.
```

Comment:
Will it be fine to just leave these as compile constants? We should avoid global states

Reply:
I don't know many ways for having compile time constants, so I added a new switch "TI_WITH_DX11_DEBUG", wondering if there is some better alternative?

Reply:
You can use `constexpr bool kD3d11DebugEnabled = false` and then this variable will become compile time constant, and branches using this variable will be compiled out

Reply:
I see, my intent is some unit tests are enabled only when the flag is true (otherwise the D3D info queue will not contain any information). This may be a user-facing feature (for example if the user wants to capture a trace with tools like RenderDoc?) but I'm not sure if Taichi is intended to be used in that way so I'm wondering which of the following make sense:
* It is set to `constexpr bool kD3d11DebugEnabled = false` in the .cpp file. Anyone who wants to enable it needs to modify the code manually.
* The flag and the test is controlled by the TI_WITH_DX11_DEBUG flag.
* Or something else?

Reply:
The unit tests should run if the backend is built & tests are built. The unit test and debug layer should only be used for developers, so leaving it in the code is fine.

Reply:
Thanks for the explanation, I removed the extra flag and left `kD3d11DebugEnabled` in the code.